littleX_BE/littleX.jac

```diff
@@ -0,0 +1,230 @@
+
+node profile {
+    has username: str = "";
+
+    can update with update_profile entry {
+        logger.info("Updating profile");
+        self.username = here.new_username;
+        report self;
+    }
+
+    can get with get_profile entry {
+        logger.info(f"Profile info: {self}");
+        report self;
+    }
+
+    can follow with follow_request entry {
+        logger.info(f"New followee addded: {here.profile_id}");
+        self +:follow():+> &here.profile_id;
+        report self;
+    }
+
+    can un_follow with un_follow_request entry {
+        logger.info(f"Followee removed: {here.profile_id}");
+        self del-:follow:-> &here.profile_id;
+        report self;
+    }
+
+    can like_tweet with like_tweet entry {
+        tweet_node = &here.tweet_id;
+        tweet_node <+:like():+> self;
+        logger.info(f"Liked a tweet: {tweet_node}");
+        report tweet_node;
+    }
+
+    can remove_like with remove_like entry {
+        tweet_node = &here.tweet_id;
+        tweet_node del-:like:-> self;
+        logger.info(f"Removed like from tweet: {tweet_node}");
+        report tweet_node;
+    }
+}
+
+node tweet {
+    has content: str;
+    has created_at: str = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S");
+    has tweet_info: dict = {};
+
+    can update with update_tweet exit {
+        self.content = here.updated_content;
+        logger.info(f"Tweet updated: {self}");
+        report self;
+    }
+
+    can delete with remove_tweet exit {
+        logger.info(f"Tweet deleted: {self}");
+        del self;
+    }
+
+    can load with load_tweets entry {
+        logger.info(f"Tweet loaded: {here}");
+        self.tweet_info = {
+            "id": jid(self),
+            "content": self.content,
+            "likes": [i.username for i in [self-:like:->]],
+            "comments": [{"id": jid(i), "content": i.content} for i in [self-->(`?comment)]]
+        };
+    }
+}
+
+node comment {
+    has content: str;
+}
+
+edge follow {}
+
+edge like {}
+
+edge post {}
+
+walker user_init {
+    can user_init with `root entry {
+        visit [-->(`?profile)] else {
+            logger.info("Creating new profile");
+            new_profile = here ++> profile();
+            Jac.unrestrict(new_profile[0], level="CONNECT");
+            visit new_profile;
+        }
+    }
+}
+
+walker update_profile :user_init: {
+    has new_username: str;
+}
+
+walker get_profile  :user_init: {}
+
+walker load_user_profiles {
+    obj __specs__ {
+        static has auth: bool = False;
+    }
+    can load_profiles with `root entry {
+        self.profiles: list = [];
+        
+        for user in NodeAnchor.Collection.find({"name": "profile"}) {
+            user_node = user.architype;
+            self.profiles.append(
+                {"name": user_node.username, "id": jid(user_node)}
+            );
+        }
+        report self.profiles;
+    }
+}
+
+walker follow_request :user_init: {
+    has profile_id: str;
```

Comment:
Directly spawn on profile node. (similar to what we talked about for like/comment)